src/square.py

```diff
@@ -78,20 +78,38 @@ class Square(Vector):
 
     notation_range = re.compile(f"[{file_range}][{rank_range}]")
 
+    def __new__(cls, square: str | Vector):
+        """Abort if attempted square is illegal.
+
+        This is used to make `Square` objects self-correcting upon behavior.
+        Failure of creation will be its own validity check.
+
+        Args:
+            square: A square in notation form or a vector (assuming the board origin is top-left in the latter case).
+
+        Returns:
+            `None` if square specification is illegal.
+        """
+        try:
+            if isinstance(square, str):
+                assert Square.notation_range.match(square)
+            if isinstance(square, Vector):
+                assert 0 <= square.rank < 8 and 0 <= square.file < 8
+            return super(Square, cls).__new__(cls)
+        except AssertionError:
+            return None
```

Comment:
Catching assertion error to return `None` based on some truth value testing is kind of iffy. Α more direct way is
```python
       valid = False
        if isinstance(square, str):
            valid = Square.notation_range.match(square)
        elif isinstance(square, Vector):
            valid = square.rank in range(8) and square.file in range(8)
        if valid:
            return super().__new__(cls)
        return None
```

Reply:
Agreed, this will be replaced. Commentary: `super` in class methods has to be explicitly be called with arguments, I think. I'll try again without arguments and see what happens.